daemon/daemon_main.go

```diff
@@ -1085,6 +1088,17 @@ func initEnv(cmd *cobra.Command) {
 		}
 	}
 
+	if option.Config.EnableDSR {
+		switch {
+		case !option.Config.EnableNodePort:
+			log.Fatal("DSR can be enabled only for NodePort BPF (--enable-node-port)")
+		case option.Config.EnableDSR && option.Config.EnableIPv6:
+			log.Fatal("DSR cannot be used with IPv6")
+		case option.Config.Tunnel != option.TunnelDisabled:
+			log.Fatal("DSR cannot be used with tunnel")
```

Comment:
Why is tunneling not supported? Couldn't the backend do the same logic here as in, sending the reply directly back (not via tunnel dev but phys dev)?

Reply:
Discussed offline: I will do it as a follow-up (v1.8).